src/device.rs

```diff
@@ -12,19 +12,38 @@ use wgc::{gfx_select, id};
 #[no_mangle]
 pub unsafe extern "C" fn wgpuInstanceRequestAdapter(
     _: native::WGPUInstance,
-    options: *const native::WGPURequestAdapterOptions,
+    options: &native::WGPURequestAdapterOptions,
     callback: native::WGPURequestAdapterCallback,
     userdata: *mut std::os::raw::c_void,
 ) {
-    let compatible_surface: Option<id::SurfaceId> =
-        options.as_ref().map(|options| options.compatibleSurface);
+    let (compatible_surface, given_backend_bits) = follow_chain!(
+        map_adapter_options(options,
+        WGPUSType_AdapterExtras => native::WGPUAdapterExtras)
+    );
+    // Convert backend bits. We need to initialize with *something*, thus the xor at the end.
+    let mut backend_bits = wgt::BackendBit::VULKAN;
+    if given_backend_bits & native::WGPUBackendBits_Metal as u32 > 0 {
```

Comment:
where are we defaulting to PRIMARY? We should be defaulting if the user doesn't pass the extra structure

Reply:
Fixed.